source-reminder/lambda_function.py

```diff
@@ -0,0 +1,146 @@
+import gspread
+from oauth2client.service_account import ServiceAccountCredentials
+import requests
+from datetime import date, datetime, timedelta
+import logging
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+scope = ['https://spreadsheets.google.com/feeds', 'https://www.googleapis.com/auth/drive', 'https://www.googleapis.com/auth/admin.directory.user']
+API_ENDPOINT = 'https://www.texastribune.org/api/v2/articles/'
+credentials = ServiceAccountCredentials.from_json_keyfile_name('credentials.json', scope)  # Add credentials to the service account.
+client = gspread.authorize(credentials)  # Authorize the client sheet.
```

Comment:
One teency nit for code readability is that I think the code comments would read easier if they were above the line of code they're commenting on.  I think vertical spacing is a friend for the 👀 .  Also appreciate how well-commented this all is 😌